### DISCUSSION TITLE
Getting Time Display

### DISCUSSION THREAD

**Discussion by Vahen1981** [tttapa/Control-Surface#497] [Q&A] [answered]
Hi Pieter!
I´m trying to print the TimeDisplay in serial monitor (I also tried in a lcd), but I can´t do it... Could you please tell me what I'm doing wrong??
I used this code:

```
#include <Control_Surface.h>
USBMIDI_Interface usbmidi;



MCU::TimeDisplay timedisplay = {};

void showTime(){
  char text[10];
  timedisplay.getBars(text);
  Serial.println(text);
}


void setup() {
  Control_Surface.begin();
  Serial.begin(115200);
}


void loop() {
  Control_Surface.loop();
  showTime();
  
}
```

Thank you!!!

**Comment by tttapa** [comment, marked as answer]
You might want to try `getText()` first, `getBars()` is not that smart and different DAWs may use a different format which might confuse it.

```cpp
 void showTime() {
  Serial.println(timedisplay.getText());
}
```

**Reply by Vahen1981** [reply]
When I try that I get a compiler error:
time_display:15: error: no matching function for call to 'CS::MCU::TimeDisplay::getText()

What could be?

**Reply by tttapa** [reply]
Ah my bad, I was thinking about the `LCDDisplay` class, for `TimeDisplay` you could try this https://tttapa.github.io/Control-Surface-doc/Doxygen/df/d69/classMCU_1_1TimeDisplay.html#aab5b68fd9d99cdb8970e8d44ccefe20e:
```cpp
void showTime() {
  char text[MCU::TimeDisplayLength + 1]; // +1 for null terminator
  timedisplay.getText(text);
  Serial.println(text);
}
```

**Reply by Vahen1981** [reply]
I tried that, but I received just empty spaces in the serial monitor... Maybe is my Daw? I'm using Cubase

**Reply by tttapa** [reply]
Maybe Cubase uses SysEx to update the time display. Control Surface currently only supports CC for the time display. You should probably have at the messages sent by Cubase in a MIDI monitor, if you know the type of messages it sends I can tell you how to receive them.

**Reply by Vahen1981** [reply]
Ok, when I use the "Mackie-Control-Universal-Reverse-Engineering" i get this kind of messages in the serial monitor:

bf 42 31 on cable 1
bf 40 37 on cable 1
bf 41 31 on cable 1
bf 40 38 on cable 1
bf 41 31 on cable 1
bf 42 20 on cable 1
bf 43 72 on cable 1
bf 40 37 on cable 1
bf 41 32 on cable 1
90 5d 7f	(STOP) on cable 1

**Reply by tttapa** [reply]
It's sending the time display data on channel 16, try changing the channel in your constructor.

**Reply by Vahen1981** [reply]
Great! Now is working... thank you very much!
So, I did this:

```
MCU::TimeDisplay timedisplay = {CHANNEL_16};


void showTime() {
  char text[MCU::TimeDisplayLength + 1]; // +1 for null terminator
  timedisplay.getText(text, 0, MCU::TimeDisplayLength);
  Serial.println(text);
}
```
